src/Sylius/Bundle/ApiBundle/CommandHandler/Cart/RemoveItemFromCartHandler.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\CommandHandler\Cart;
+
+use Sylius\Bundle\ApiBundle\Command\Cart\RemoveItemFromCart;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\Component\Order\Modifier\OrderModifierInterface;
+use Sylius\Component\Order\Processor\OrderProcessorInterface;
+use Sylius\Component\Order\Repository\OrderItemRepositoryInterface;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Webmozart\Assert\Assert;
+
+final class RemoveItemFromCartHandler implements MessageHandlerInterface
+{
+    /** @var OrderRepositoryInterface */
+    private $orderRepository;
+
+    /** @var OrderItemRepositoryInterface */
+    private $orderItemRepository;
+
+    /** @var OrderModifierInterface */
+    private $orderModifier;
+
+    /** @var OrderProcessorInterface */
+    private $orderProcessor;
+
+    public function __construct(
+        OrderRepositoryInterface $orderRepository,
+        OrderItemRepositoryInterface $orderItemRepository,
+        OrderModifierInterface $orderModifier,
+        OrderProcessorInterface $orderProcessor
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->orderItemRepository = $orderItemRepository;
+        $this->orderModifier = $orderModifier;
+        $this->orderProcessor = $orderProcessor;
+    }
+
+    public function __invoke(RemoveItemFromCart $removeItemFromCart): OrderInterface
+    {
+        /** @var OrderItemInterface|null $orderItem */
+        $orderItem = $this->orderItemRepository->findOneByIdAndCartTokenValue(
+            $removeItemFromCart->orderItemId,
+            $removeItemFromCart->tokenValue
+        );
+
+        Assert::notNull($orderItem);
+
+        /** @var OrderInterface|null $cart */
+        $cart = $this->orderRepository->findOneBy([
+            'state' => OrderInterface::STATE_CART,
+            'tokenValue' => $removeItemFromCart->tokenValue,
+        ]);
+
+        Assert::notNull($cart);
```

Comment:
Instead of fetching cart from the repository, we can just check if order related to order item has expected token I suppose
```suggestion
        $cart = $orderItem->getOrder();

        Assert::same($cart->getTokenValue(), $removeItemFromCart->tokenValue);
```